### ISSUE TITLE
Alternative to goog.ds?

### ISSUE THREAD

**Issue by drewkutilek** [google/closure-library#1178]
Admittedly, my project is on a very old version of the `closure-library`, but when I went back to look at the auto-generated documentation I noticed that that `goog.ds` module was not there. Further digging lead me to this commit: https://github.com/google/closure-library/commit/49f6e6c90334604641e3da784a541162984382a9

Which then lead me to these release notes that detail the modules removal:
https://github.com/google/closure-library/releases/tag/v20210302

What was the reason for the removal? Is there a suggested alternative? I ask because my project uses the `DataManager` class extensively to store data and notify on changes.

**Comment by concavelenz** [comment]
Is there a reason you can't fork the code into your own project if you are using them?

**Comment by 12wrigja** [comment]
`goog.ds` was removed as it was unused within Google. There is no suggested alternative, but John's suggestion of copying the newest version of it into your project is a good one.

**Comment by drewkutilek** [comment]
Yes I agree that copying the latest version into my project is a viable alternative, but I was more curious as to why it became unused and if there is a compelling reason for me to move away from it as well. I found it very useful.

**Comment by shicks** [comment]
I'm not particularly familiar with this library myself, but I did a little bit of archaeology.  It seems that the package was relatively popular 10ish years ago, but apparently the uses were replaced by other things, or else just deleted as projects retired.  By a last year, we realized there was only a single user left within Google, so we handed ownership off to them.

It's possible something like redux might be a more modern alternative, but I can't speak to whether it covers all the use cases.  The easiest solution is definitely to just fork it.
